src/components/tl/dpu/tl_dpu_context.c

```diff
@@ -0,0 +1,246 @@
+            tl_dpu_config->use_dpu = 0;
+        }
+        free(h);
+    }
+    else {
+        goto err;
+    }
+
+    if (!dpu_found) {
+        goto err;
+    }
+
+    fprintf(stdout, "Connecting to %s", hname);
+    sockfd = _server_connect(hname, tl_dpu_config->server_port);
+
+    memset(&ucp_params, 0, sizeof(ucp_params));
+    ucp_params.field_mask      = UCP_PARAM_FIELD_FEATURES |
+                                 UCP_PARAM_FIELD_REQUEST_SIZE |
+                                 UCP_PARAM_FIELD_REQUEST_INIT |
+                                 UCP_PARAM_FIELD_REQUEST_CLEANUP;
+    ucp_params.features        = UCP_FEATURE_TAG |
+                                 UCP_FEATURE_RMA;
+    ucp_params.request_size    = sizeof(tl_dpu_ucp_request_t);
+    ucp_params.request_init    = tl_dpu_ucp_req_init;
+    ucp_params.request_cleanup = tl_dpu_ucp_req_cleanup;
+
+    status = ucp_init(&ucp_params, NULL, &ucp_context);
+    if (status != UCC_OK) {
+        fprintf(stderr, "failed ucp_init(%s)\n", ucc_status_string(status));
+        goto err;
+    }
+
+    memset(&worker_params, 0, sizeof(worker_params));
+    worker_params.field_mask    = UCP_WORKER_PARAM_FIELD_THREAD_MODE;
+    worker_params.thread_mode   = UCS_THREAD_MODE_SINGLE;
+    status = ucp_worker_create(ucp_context, &worker_params, &ucp_worker);
+     if (status != UCC_OK) {
+        fprintf(stderr, "failed ucp_worker_create (%s)\n", ucc_status_string(status));
+        goto err_cleanup_context;
+    }
+
+    worker_attr.field_mask = UCP_WORKER_ATTR_FIELD_ADDRESS |
+                             UCP_WORKER_ATTR_FIELD_ADDRESS_FLAGS;
+    worker_attr.address_flags = UCP_WORKER_ADDRESS_FLAG_NET_ONLY;
+    ucp_worker_query(ucp_worker, &worker_attr);
+
+    ret = send(sockfd, &worker_attr.address_length, sizeof(&worker_attr.address_length), 0);
+    if (ret<0) {
+        fprintf(stderr, "send length failed");
+    }
+    ret = send(sockfd, worker_attr.address, worker_attr.address_length, 0);
+    if (ret<0) {
+        fprintf(stderr, "send address failed");
+    }
+    ret = recv(sockfd, &rem_worker_addr_size, sizeof(rem_worker_addr_size), MSG_WAITALL);
+    if (ret<0) {
+        fprintf(stderr, "recv address length failed");
+    }
+    rem_worker_addr = malloc(rem_worker_addr_size);
+    ret = recv(sockfd, rem_worker_addr, rem_worker_addr_size, MSG_WAITALL);
+    if (ret<0) {
+        fprintf(stderr, "recv address failed");
+    }
+    ep_params.field_mask       = UCP_EP_PARAM_FIELD_FLAGS |
+                                 UCP_EP_PARAM_FIELD_REMOTE_ADDRESS |
+                                 UCP_EP_PARAM_FIELD_ERR_HANDLER |
+                                 UCP_EP_PARAM_FIELD_ERR_HANDLING_MODE;
+    ep_params.err_mode         = UCP_ERR_HANDLING_MODE_PEER;
+    ep_params.err_handler.cb   = err_cb;
+    ep_params.address          = rem_worker_addr;
+
+    status = ucp_ep_create(ucp_worker, &ep_params, &ucp_ep);
```

Comment:
You local var "status" is ucc_status while ucp_ep_create returns ucs_status_t. Use usc_status_to_ucc helper function from utils.